File: source/phpfiles/utils/weha/diff/SentenceMatch.php
<?php
include_once('SentenceEdit.php');

class SentenceMatch extends SentenceEdit {

	private $oldOrder;
	private $newOrder;

	public function __construct($oldPos, $newPos, $oldS, $newS, $mr) {
		$this->oldPos = $oldPos;
		$this->newPos = $newPos;
		$this->oldSentence = $oldS;
		$this->newSentence = $newS;
		$this->matchingRate = $mr;
		$this->oldOrder = -1;
		$this->newOrder = -1;
	}

	public function descString() {
		return ("SentenceMatch " . $this->oldSentence . " " . $this->newSentence);
	}

	public function getOldOrder() {
		return $this->oldOrder;
	}

	public function setOldOrder($oldOrder) {
		$this->oldOrder = $oldOrder;
	}

	public function getNewOrder() {
		return $this->newOrder;
	}

	public function setNewOrder($newOrder) {
		$this->newOrder = $newOrder;
	}
}

File: source/weha/wikianalysis/diff/sentence/SentenceMatch.java
package mo.umac.wikianalysis.diff.sentence;

public class SentenceMatch extends SentenceEdit {

	private int oldOrder;
	private int newOrder;

	public SentenceMatch(int oldPos, int newPos, 
			Sentence oldS, Sentence newS, double mr) {
		super();
		this.oldPos = oldPos;
		this.newPos = newPos;
		this.oldSentence = oldS;
		this.newSentence = newS;
		this.matchingRate = mr;
		this.oldOrder = -1;
		this.newOrder = -1;
	}

	public SentenceMatch(SentenceMatch m) {
		super();
		this.oldPos = m.oldPos;
		this.newPos = m.newPos;
		this.oldSentence = m.oldSentence;
		this.newSentence = m.newSentence;
		this.matchingRate = m.matchingRate;
		this.oldOrder = -1;
		this.newOrder = -1;
	}
	
	@Override
	public String descString() {
		return "SentenceMatch(" + this.oldPos+ ", " + this.newPos + ", " + this.matchingRate + ")";
	}

	public int getOldOrder() {
		return oldOrder;
	}

	public void setOldOrder(int oldOrder) {
		this.oldOrder = oldOrder;
	}

	public int getNewOrder() {
		return newOrder;
	}

	public void setNewOrder(int newOrder) {
		this.newOrder = newOrder;
	}
}
